### ISSUE TITLE
[Task]: Add documentation for the Go Direct runner.

### ISSUE THREAD

**Issue by lostluck** [apache/beam#21904]
### What would you like to happen?

https://beam.apache.org/documentation/runners/direct/  should have Go specific content for the Go direct runner.

### Issue Priority

Priority: 2

### Issue Component

Component: sdk-go

**Comment by kennycontreras** [comment]
Hi! I was thinking of working on this, but first I would like to clarify something. Currently, there are no parallelism options and streaming while using DirectRunner in Go? Therefore should those items be omitted?

**Comment by gayatrikate04** [comment]
.take-issue

**Comment by liferoad** [comment]
I do not think we need this any more. Please use https://beam.apache.org/documentation/runners/prism/
